### PR TITLE
Grau 4

### PR DESCRIPTION
**Pull request by DanMaguireData**

### CODE REVIEW

File: grau/blueprints/user_stats/routes.py [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+@user_stats_api.route("/get_user_stat", methods=["GET"])
+@login_required
+def get_user_stat():
+    """
+    Get user_stats endpoint for user_stats.
+    """
+    stats_content = request.json["user_stats_dict"]
+    return functions.get_user_stat(
+        db_session=current_session,
+        user_id=session.get("_user_id"),
```
Comment (DanMaguireData) [inline grau/blueprints/user_stats/routes.py:48]: Update

File: tests/grau/blueprints/user_stats/test_functions.py [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+        """
+        # Given
+        user_stat = self.user_stat.copy()
+        user_stat["user_id"] = self.user_id_no_stats
+        # When
+        request_response = functions.update_user_stat(db_session, user_stat)
+        # then
+        assert request_response == ("User stat not found", 404)
+
+    @pytest.mark.usefixtures("frozen_datetime")
```
Comment (DanMaguireData) [inline tests/grau/blueprints/user_stats/test_functions.py:141]: Revert back to decorator

File: tests/grau/blueprints/user_stats/test_routes.py [hunk 3]
```diff
@@ -0,0 +1,206 @@
...
+        self.current_id: int = 1
+
+    def get_user(self, **kwargs) -> Dict:
+        """
+        Returns a user dictionary.
+        """
+        user_id = self.current_id
+        self.current_id += 1
+        user = {
+            "id": user_id,
```
Comment (DanMaguireData) [inline tests/grau/blueprints/user_stats/test_routes.py:23]: Check if this is automatically created

**Review by DanMaguireData — COMMENTED** [review]
Get tests working + mereg in dans changes

File: grau/utils.py [hunk 4]
```diff
@@ -1,20 +1,29 @@
 import os
 from enum import Enum
-from typing import Optional
+from typing import Any, Optional, Type
 
 from cryptography.fernet import Fernet
 
 
-def get_secret_key():
+def get_secret_key() -> Optional[str]:
```
Comment (DANLENEHAN) [inline grau/utils.py:8]: Is this Optional because the env var could be missing?

File: grau/db/user_stats/user_stats_model.py [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+    note: Mapped[str] = mapped_column(String(100), nullable=True)
+
+    user = relationship("User", back_populates="user_stats")
+
+    def __repr__(self) -> str:
+        return (
+            f"UserStats(id={self.id!r},"
+            "value={self.value!r}, unit={self.unit!r},"
+            "created_at={self.created_at!r},"
+            "updated_at={self.updated_at!r}, note={self.note!r})"
```
Comment (DANLENEHAN) [inline grau/db/user_stats/user_stats_model.py:31]: Would we want a note in the repr? These should really just be an id or name IMO

File: grau/utils.py (hunk 4 above)
Comment (DanMaguireData) [reply grau/utils.py:8]: Yeah, mypy doesn't like that we say it returns a String when the env variable might not exist

File: grau/db/user_stats/user_stats_model.py (hunk 5 above)
Comment (DanMaguireData) [reply grau/db/user_stats/user_stats_model.py]: Removed everything from REPR but stat and user IDs

**Review by DANLENEHAN — APPROVED** [review]

**Pull request merged**